### PR TITLE
Transfer glossary from gloss.md in bc repo to reference.md (#15)

### PR DESCRIPTION
**Pull request by dhaine**

### CODE REVIEW

File: reference.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+---
+
+# Basic Operation
+
+- `# this is a comment in R`
+- Use `x <- 3` to assign a value, `3`,  to a variable, `x`
+- R counts from 1, unlike many other programming languages (e.g., Python)
+- `length(thing)` produces the length of a collection
+- `c(value1, value2, value3)` creates a vector
+- `container[i]` selects the i'th element from the container
```
Comment (fmichonneau) [inline reference.md]: from the variable `container`

File: reference.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+title: Programming with R
+subtitle: Reference
+---
+
+# Basic Operation
+
+- `# this is a comment in R`
+- Use `x <- 3` to assign a value, `3`,  to a variable, `x`
+- R counts from 1, unlike many other programming languages (e.g., Python)
+- `length(thing)` produces the length of a collection
```
Comment (fmichonneau) [inline reference.md]: returns the number of elements contained in the variable `collection`

File: reference.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+Remove objects in current environment
+`rm(x)`
+
+Remove all objects from current environment
+`rm(list = ls())`
+
+# Control Flow
+
+- Create a contitional using `if`, `elif`, and `else`
```
Comment (fmichonneau) [inline reference.md]: `elif` --> `else if`

File: reference.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+		2
+		3
+		4
+		5
+
+
+- Use `==` to test for equality
+  - `3 == 3`, will return `TRUE`,
+  - `'apple' == 'orange'` will return `FALSE`
+- `X & Y` is `True` is both X and Y are true
```
Comment (fmichonneau) [inline reference.md]: `True` --> `TRUE`

File: reference.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+- `X | Y` is `True` if either X or Y, or both are true
```
Comment (fmichonneau) [inline reference.md]: `True` --> `TRUE`

File: reference.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+# Functions
+
+- Defining a function:
+
+		is_positive <- function(integer_value){
+			if(interver_value > 0){
```
Comment (fmichonneau) [inline reference.md]: interver_value --> integer_value

**Comment by jdblischak** [comment]
Thanks, @dhaine. This is great. I noticed just two things:

1. The [format][] for the glossary has changed, so you'll need to remove the html from the old definitions.
2. The index no longer has a link to the reference sheet. Could you please add a link to this below the list of lessons in `index.md` and then render it?


[format]: https://github.com/swcarpentry/lesson-template/blob/gh-pages/LAYOUT.md#reference-guide

**Comment by jdblischak** [comment]
Great. Thanks for the PR @dhaine and thanks for the review @fmichonneau.

**Pull request merged**